### PR TITLE
Use DirectoryReader instead of IndexReader in AggregatorTestCase

### PR DESCRIPTION
**Pull request by iverase**
The method AggregatorTestCase#newIndexSearcher takes as an input an index reader. Still when running aggregations in productions we use the subclass `ContextIndexSearcher` which only supports DirectoryReader as input. Therefore I propose here that we only accept DirectoryReader as this is the only format we actually support.

There are only a few places where we were creating `MultiReader` instances. I could not really see the benefit and modify the code slightly so everything is written to the same directory. This change will help moving to support concurrency as we will probably need to be using  `ContextIndexSearcher` for that.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/AvgAggregatorTests.java [hunk 1]
```diff
@@ -265,16 +263,11 @@ public void testSingleValuedFieldPartiallyUnmapped() throws IOException {
...
         indexWriter.addDocument(singleton(new NumericDocValuesField("number", 3)));
+        indexWriter.addDocument(singleton(new NumericDocValuesField("unrelated", 100)));
         indexWriter.close();
 
-        Directory unmappedDirectory = newDirectory();
-        RandomIndexWriter unmappedIndexWriter = new RandomIndexWriter(random(), unmappedDirectory);
-        unmappedIndexWriter.close();
-
-        IndexReader indexReader = DirectoryReader.open(directory);
-        IndexReader unamappedIndexReader = DirectoryReader.open(unmappedDirectory);
```
Comment (iverase) [inline server/src/test/java/org/elasticsearch/search/aggregations/metrics/AvgAggregatorTests.java:275]: I am very confused in what this is trying to test

**Review by not-napoleon — APPROVED** [review]
I think this looks good.  I am concerned we lose a bit of coverage around fields not being present in an index, which has been a source of bugs before, but I don't think the unit tests are the most appropriate place to test that.  I'm for merging this and I'll open up an issue for making sure we have good unmapped coverage somewhere else.  It's not like the handful of places this PR changes that are comprehensive anyway.

File: test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java [hunk 2]
```diff
@@ -912,21 +873,28 @@ protected static DirectoryReader wrapInMockESDirectoryReader(DirectoryReader dir
...
+            IndexSearcher.getDefaultQueryCache(),
+            IndexSearcher.getDefaultQueryCachingPolicy(),
+            randomBoolean()
+        );
+        // if (randomBoolean()) {
+        // // this executes basic query checks and asserts that weights are normalized only once etc.
+        // return new AssertingIndexSearcher(random(), indexReader);
+        // } else {
+        // return new IndexSearcher(indexReader);
+        // }
```
[the remark this answers is not in the document]
Comment (iverase) [reply test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java]: Actually I want to keep the old one. I want to make sure we don't miss anything important with `AssertingIndexSearcher` and update the javadocs accordingly.

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/AvgAggregatorTests.java [hunk 3]
```diff
@@ -265,16 +263,11 @@ public void testSingleValuedFieldPartiallyUnmapped() throws IOException {
         indexWriter.addDocument(singleton(new NumericDocValuesField("number", 7)));
         indexWriter.addDocument(singleton(new NumericDocValuesField("number", 2)));
         indexWriter.addDocument(singleton(new NumericDocValuesField("number", 3)));
+        indexWriter.addDocument(singleton(new NumericDocValuesField("unrelated", 100)));
```
[the remark this answers is not in the document]
Comment (not-napoleon) [reply server/src/test/java/org/elasticsearch/search/aggregations/metrics/AvgAggregatorTests.java:266]: Thank you!

**Pull request merged**